lib/main.dart

```diff
@@ -47,12 +47,12 @@ void main(List<String> args) async {
     String version;
     String channel;
 
-    if (sdk.endsWith('.x')) {
+    if (sdk.split('.').length == 2) {
+      // Handle the wildcard (`2.19`, `3.1`, ...) format.
       channel = 'stable';
 
-      // Find the latest version for 'sdk'.
-      final versionPrefix = sdk.substring(0, sdk.length - '.x'.length);
-      version = await findLatestSdkForRelease(versionPrefix);
+      // Find the latest version for the given sdk release.
+      version = await findLatestSdkForRelease(sdk);
```

Comment:
How about the same but for beta releases? E.g. the latest version in a beta series? Or maybe 3.3.0-1.beta would give you 3.3.0-1.1.beta and later 3.3.0-1.2.beta and onwards.

Although in those cases you can kinda just follow the beta channel, although that would give potential breakages whenever a new beta series is dropped, whereas a patched beta is only more stable.

Reply:
Do you mean something like `3.1-beta`? I.e., the latest beta for a given sdk release?

I don't think that's a common use case. I think testing against the latest `stable` is common, and testing against the upcoming releases (`beta`, `dev`, ...) is common. What this PR will help with is for people that want to test against specific SDKs (for example, the lower sdk constraint of their pubspec), but want the latest patch release for that SDK. Right now they're using `3.1.0` as a stand-in for that lower bounds, but I think they really want to test against `3.1.<latest patch>`.